src/tests/sqlsmith/src/sqlreduce/path.rs

```diff
@@ -0,0 +1,1035 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Path-based AST navigation for SQL reduction.
+//!
+//! This module provides utilities for navigating and modifying SQL ASTs using
+//! path-based addressing for precise AST manipulation.
+
+use std::fmt;
+
+use risingwave_sqlparser::ast::*;
+
+/// Represents a path component in an AST navigation path.
+/// Components that make up a path through the AST.
+/// Enables precise navigation to any AST node.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PathComponent {
+    /// Field access by name (e.g., "selection", "projection")
+    Field(String),
```

Comment:
Instead of using `String`, which is error prone, why don't we use an `enum` instead?